feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/view/AlertsView.kt

```diff
@@ -0,0 +1,64 @@
+package jp.co.soramitsu.feature_staking_impl.presentation.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.LinearLayout
+import androidx.annotation.StringRes
+import androidx.recyclerview.widget.LinearLayoutManager
+import jp.co.soramitsu.common.utils.makeGone
+import jp.co.soramitsu.common.utils.makeVisible
+import jp.co.soramitsu.common.view.shape.getCutCornerDrawable
+import jp.co.soramitsu.feature_staking_impl.R
+import jp.co.soramitsu.feature_staking_impl.presentation.staking.alerts.model.AlertModel
+import jp.co.soramitsu.feature_staking_impl.presentation.staking.alerts.AlertsAdapter
+import kotlinx.android.synthetic.main.view_alert.view.*
+
+class AlertsView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+) : LinearLayout(context, attrs, defStyle), AlertsAdapter.ItemHandler {
+
+    sealed class Status {
+        class Alerts(val alerts: List<AlertModel>) : Status()
+
+        object NoAlerts : Status()
+    }
```

Comment:
Do you plan to have mirrored class on ViewModel's layer and mapper between them?

If no, I suggest moving these classes out of AlertView, since it will be used by ViewModel's and ViewModels should know nothing about Views (rather just provide data to display)

Reply:
Done